pubsub/pubsub_goredis.go

```diff
@@ -109,11 +110,11 @@ func (ps *GoRedisPubSub) Publish(ctx context.Context, topic, message string) err
 	return ps.client.Publish(ctx, topic, message).Err()
 }
 
-func (ps *GoRedisPubSub) Subscribe(ctx context.Context, topic string) Subscription {
+func (ps *GoRedisPubSub) Subscribe(ctx context.Context, topic string, callback func(string)) Subscription {
 	sub := &GoRedisSubscription{
 		topic:  topic,
 		pubsub: ps.client.Subscribe(ctx, topic),
```

Comment:
It looks like goredis doesn't keep track of existing topics. That means if the same topic is subscribed multiple times, the same piece of message will be delivered through the network multiple times. For Refinery's use case, that probably is ok since we don't really have multiple subscribers for the same topic as far as I know. Maybe we can add a comment here so that it's recorded?

Reply:
I've added the comment, but this is why I had originally created the Topic concept as a separate entity, and we decided to remove it.